hal/ia32/_init.S

```diff
@@ -180,23 +148,21 @@ _start_initap:
 	subl $VADDR_KERNEL, %eax
 	movl %eax, %esi
 	movl $0x200, %ecx
-	cld;
-	rep; movsb
-
-//jmp b3
+	cld
+	rep movsb
 
 	/* Send INI IPI */
-	movl $0x000c4500, %eax /* 11 00 0 1 0 0 0 101 00000000 */;
-	mov %eax, (0xfee00300)
-b0:
-	btl $12, (0xfee00300)
-	jc b0
+	pushl $0x000c4500 /* 11 00 0 1 0 0 0 101 00000000 */
+	pushl 0xff
+	call cpu_sendIPI
+	addl $8, %esp
 
 	/* Send SIPI IPI (with real memory-address 0x00011000) */
-	movl $0x000c4611, %eax /* 11 00 0 0 1 0 0 0 110 00010001 */
-	mov %eax, (0xfee00300)
-b1:	btl $12, (0xfee00300)
-	jc b1
+	pushl $0x000c4611 /* 11 00 0 0 1 0 0 0 110 00010001 */
+	pushl 0xff
+	call cpu_sendIPI
```

Comment:
cpu_sendIPI has been removed from API - a prototype should be provided in this file or in hal/cpu.h (imho better in the headers)